### PR TITLE
chore(functions/concepts): Migrate Send HTTP requests Sample to gen2

### PR DESCRIPTION
**Pull request by KAKAinGoog**
Migrate Send HTTP requests Sample to gen2

### CODE REVIEW

File: functions/concepts/requests/index.js [hunk 1]
```diff
@@ -16,16 +16,18 @@
...
  * @param {Object} req Cloud Function request context.
  * @param {Object} res Cloud Function response context.
  */
-exports.makeRequest = async (req, res) => {
+functions.http('makeRequest', (req, res)  => {
   const url = 'https://example.com'; // URL to send the request to
-  const externalRes = await fetch(url);
-  res.sendStatus(externalRes.ok ? 200 : 500);
-};
+  fetch(url).then((externalRes) => {
```
Comment (ace-n) [inline functions/concepts/requests/index.js:29]: Can we revert this change? `await` is considered cleaner than `.then`.

**Review by ace-n — CHANGES_REQUESTED** [review]

File: functions/concepts/requests/index.js (hunk 1 above)
Comment (KAKAinGoog) [reply functions/concepts/requests/index.js]: Done

**Review by ace-n — APPROVED** [review]

File: functions/concepts/requests/index.js [hunk 2]
```diff
@@ -16,16 +16,17 @@
...
+const functions = require('@google-cloud/functions-framework');
 
 /**
  * HTTP Cloud Function that makes an HTTP request
  *
  * @param {Object} req Cloud Function request context.
  * @param {Object} res Cloud Function response context.
  */
-exports.makeRequest = async (req, res) => {
+functions.http('makeRequest', async (req, res)  => {
```
Comment (ace-n) [inline functions/concepts/requests/index.js:27]: This should fix the lint error
```suggestion
functions.http('makeRequest', async (req, res) => {
```

**Comment by ace-n** [comment]
All checks (except the `pull_request` ones) are passing.

➡️ time to merge
